diffpy/snmf/subroutines.py

```diff
@@ -105,10 +105,10 @@ def construct_component_matrix(components, signal_length):
     if len(components) <= 0:
         raise ValueError(f"Number of components = {len(components)}. Number_of_components must be >= 1")
 
-    component_matrix = np.zeros((signal_length, len(components)))
+    component_matrix = np.zeros((signal_length, len(components))).T
```

Comment:
I think it would be clearer if you did `component_matrix = np.zeros((len(components), signal_length))` and then returned `component_matrix`, no?

Reply:
Yes. `component_matrix` would have the incorrect shape, so you would have to transpose it.

Reply:
even after you change the order in the np.zeros as I did?

Reply:
`component_matrix` should have the shape `signal_length` by `number_of_components` once it is returned. If I had ` component_matrix = np.zeros((len(components), signal_length))` I think that I would have to transpose it after I have replaced the rows inside the for loop.